packages/yew/src/functional/hooks/use_effect.rs

```diff
@@ -2,11 +2,25 @@ use std::cell::RefCell;
 
 use crate::functional::{hook, Effect, Hook, HookContext};
 
+/// Trait describing the destructor of [`use_effect`] hook.
+pub trait TearDown: Sized + 'static {
+    /// The function that is executed when destructor is called
+    fn tear_down(self) {}
```

Comment:
```suggestion
    fn tear_down(self);
```

I feel we should require the `tear_down` method to be specified by user. As a default behaviour can be covered by `()` already.